### PR TITLE
new lib change

### PR DESCRIPTION
**Pull request by Fan1117**
Signed-off-by: Fan1117 <macfan1995@gmail.com>

### CODE REVIEW

File: matlab-package/BoardShim.m [hunk 1]
```diff
@@ -28,12 +28,24 @@
             obj.exit_codes(16) = 'ANOTHER_BOARD_IS_CREATED_ERROR';
             obj.exit_codes(17) = 'GENERAL_ERROR';
             obj.exit_codes(18) = 'SYNC_TIMEOUT_ERROR';
-            if isunix
-                obj.libname = 'libBoardController';
+            
+            if ispc
```
Comment (Andrey1994) [inline matlab-package/BoardShim.m]: ispc means windows, right?

File: matlab-package/BoardShim.m [hunk 2]
```diff
@@ -28,12 +28,24 @@
...
+                    loadlibrary('lib/libBoardController.dylib','inc/board_controller.h')
+                end
+            elseif isunix
+                obj.libname = 'libBoardController';
+                if not(libisloaded('libBoardController'))
+                    loadlibrary('lib/libBoardController.so', 'inc/board_controller.h')
+                end
             else
-                obj.libname = 'BoardController';
+                disp 'OS not supported!'
```
Comment (Andrey1994) [inline matlab-package/BoardShim.m]: can we raise an exception here?

**Comment by Andrey1994** [comment]
Did you test it on all OSes?

File: matlab-package/BoardShim.m (hunk 1 above)
Comment (Fan1117) [reply matlab-package/BoardShim.m]: yes, ispc means windows in matlab.

File: matlab-package/BoardShim.m (hunk 2 above)
Comment (Fan1117) [reply matlab-package/BoardShim.m]: I have change it to error ('OS not supported!')

**Comment by Andrey1994** [comment]
thanks, looks good to me!
Did you test it? I wanna merge it after #49 so maybe some merge conflicts will occur

**Pull request merged**